docs/pbc/all/merchant-management/202407.0/marketplace/install-and-upgrade/tutorials-and-howtos/create-merchant-commission-calculator-type-plugins.md

```diff
@@ -0,0 +1,345 @@
+---
+title: Create merchant commission calculator type plugins
+description: Learn how to implement merchant commission calculator type plugin.
+template: howto-guide-template
+last_updated: Jul 22, 2024
+---
+
+This document describes how to create and register a merchant commission calculator type plugin to provide custom calculation logic for merchant commissions.
+
+Approximate time to complete: 1 hour.
+
+## Prerequisites
+
+[Install the Marketplace Merchant Commission feature](/docs/pbc/all/merchant-management/{{page.version}}/marketplace/install-and-upgrade/install-features/install-the-marketplace-merchant-commission-feature.html).
+
+
+## 1) Adjust transfer definitions
+
+To provide the required order item data, adjust the definition of the `MerchantCommissionCalculationRequestItem` transfer object. `MerchantCommissionCalculationRequestItemTransfer` is populated with data from the `spy_sales_order_item` database table. to provide the price to pay aggregation amount of the order item, add the `sumPriceToPayAggregation` property to the transfer object.
```

Comment:
Is this just one step?

Reply:
Yes, when DB and transfer properties are named the same, mapping will be done without any additional code changes.